File: 3.c
#include <stdio.h>
#include <math.h>
#include <stdint.h>
#include <string.h>

int main()
{
	int error, exit1;
	char number; 
	float F, G, Y, a, x, x2, shag;


	while (exit1 != 1)
	{
	printf("Функция G - 1, Функция F - 2, Функция Y - 3, Введите число: \n");
	scanf("%c", &number);

	switch (number)
	{
		case '1':
    	    printf ("Введите число a \n");
       		scanf ("%f", &a);
       		printf ("Введите число x \n");
       		scanf ("%f", &x);
       		printf ("Введите число x2 \n");
       		scanf ("%f", &x2);
       		printf ("Введите шаг вычисления функции\n");
       		scanf ("%f", &shag);
       		if (shag > x2)
       			{
       			printf ("Шаг вычисления функции больше чем область вычисления функции \n");
       			error = 6;
       			break;
       			}

          for (;x <= x2-shag;x = x + shag)
            {
	 			if ((-20*pow(a,2)+28*a*x+3*pow(x,2)) == 0)
              	{
                	error = 1;  
                	printf("Функция не может быть вычислена так как знаменатель не может быть = 0\n");
                	break;
              	}
              	G = 4*(-4*pow(a,2)+a*x+5*pow(x,2))/(-20*pow(a,2)+28*a*x+3*pow(x,2));
	    		printf("%f\t %f\n",G,x);
	    	}
	  break;
	   	
		break;

		case '2':
	   		printf ("Введите число a \n");
       		scanf ("%f", &a);
       		printf ("Введите число x \n");
       		scanf ("%f", &x);
       		printf ("Введите число x2 \n");
       		scanf ("%f", &x2);
       		printf ("Введите шаг вычисления функции\n");
       		scanf ("%f", &shag);

       		if (shag > x2)
       			{
       			printf ("Шаг вычисления функции > Области вычисления функции\n");
       			error = 6;
       			break;
       			}

       		for (;x <= x2-shag;x = x + shag)
       			{ 
       		     	F = (atan(24*pow(a,2)-25*a*x+6*pow(x,2)));
	    		    printf("%f\t %f\n",F,x);
	    		
       			}
		break;

		case '3':
	   		printf ("Введите число a \n");
       		scanf ("%f", &a);
       		printf ("Введите число x \n");
       		scanf ("%f", &x);
        	printf ("Введите число x2 \n");	
        	scanf ("%f", &x2);
        	printf ("Введите шаг вычисления функции\n");
       		scanf ("%f", &shag);
       		if (shag > x2)
       			{
       			printf ("Шаг вычисления функции > Области вычисления функции\n");
       			break;
       			}
       		for (x;x <= x2;x = x + shag)
       			{ 	
	    			  Y = (log(2*pow(a,2)-7*a*x+6*pow(x,2)+1));
	    			  printf("%f\t %f\n",Y,x);
        		}
		break;

		default:
			printf("Такой функции нет \n");
			error = 5;
		
		}	
	
	printf("Хотите выйти из программы? 1 - Да 0 - Нет\n");
	scanf ("%d", &exit1);
	getchar();
	}
	return error;
}

File: 3.py
import math

exit1 = 0
while exit1 == 0:

	print("Функция G - 1, Функция F - 2, Функция Y - 3, Введите число: \n")
	number = int(input())
	a = float (input ("a:"))
	x = float (input ("x:"))
	x2 = float (input("x2:"))
	shag = float (input("Шаг вычисления:"))
	if shag > x2:
		print ("Шаг вычисления функции больше чем область вычисления функции\n")
		continue

	if number == 1:
		while x <= x2:
			if (-20*pow(a,2)+28*a*x+3*pow(x,2)) == 0:
				print ("Функция не может быть вычислена так как знаменатель не может быть = 0\n")
				exit(0)
			G = 4*(-4*pow(a,2)+a*x+5*pow(x,2))/(-20*pow(a,2)+28*a*x+3*pow(x,2))
			print ('{:2f}\t {:2f}'.format(x,G))
			x = x + shag

	elif number == 2:
		a = float (input ("a:"))
		x = float (input ("x:"))
		x2 = float (input("x2:"))
		shag = float (input("Шаг вычисления:"))
		if shag > x2:
			print ("Шаг вычисления функции > Области вычисления функции\n")
			continue
		while x <= x2:	
			F = (atan(24*pow(a,2)-25*a*x+6*pow(x,2)))
			print ('{:2f}\t {:2f}'.format(x,F))
			x = x + shag
			
	elif number == 3:
			a = float (input ("a:"))
			x = float (input ("x:"))
			x2 = float (input("x2:"))
			shag = float (input("Шаг вычисления:"))
			if shag > x2:
				print ("Шаг вычисления функции > Области вычисления функции\n")
				continue
			Y = (log(2*pow(a,2)-7*a*x+6*pow(x,2)+1))
			print ('{:2f}\t {:2f}'.format(x,Y))
			x = x + shag
			
	else:
		print('Такой функции нет \n')

	print ("Хотите выйти из программы 1 - да 0 - нет")
	exit1 = int(input())		
	
